### PR TITLE
get_feature_dict with nested types

### PR DESCRIPTION
**Pull request by solbrigm**
Deal with nested types in the return: Map, Struct, and Arrays of Arrays.

### CODE REVIEW

File: tecton_client/_internal/data_types.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 from dataclasses import dataclass
 from typing import Any, Dict, List, Optional
 
+_SPECIAL_NUMBER_STRINGS = {"Infinity", "-Infinity", "NaN"}
```
Comment (brian-tecton-ai) [inline tecton_client/_internal/data_types.py:4]: These are valid responses we receive? Seem odd for an int64 right?

File: tecton_client/_internal/data_types.py [hunk 2]
```diff
@@ -45,16 +47,40 @@ def get_features_dict(self) -> Dict[str, Any]:
             for metadata, raw_feature_value in zip(features_metadata, self.result.features)
         }
 
-    def _get_feature_value(self, data_type, raw_feature_value):
+    def _get_feature_value(self, data_type_info, raw_feature_value):
```
Comment (brian-tecton-ai) [inline tecton_client/_internal/data_types.py:50]: type annotations?

Comment (brian-tecton-ai) [inline tecton_client/_internal/data_types.py:50]: This seems closer in semantics to a pure function right? We don't use `self.result` nor `self.metadata`. Let's move it out of the class and then we leave ourselves open for better unit tests

Comment (brian-tecton-ai) [reply tecton_client/_internal/data_types.py:50]: like ideally we had unit tests that just loops over a set of possible data types and json and test those directly

File: tecton_client/_internal/data_types.py [hunk 3]
```diff
@@ -45,16 +47,40 @@ def get_features_dict(self) -> Dict[str, Any]:
...
+            }
+        elif data_type == "struct":
+            field_meta_list = data_type_info.get("fields")
+            value = {
+                field_meta.get("name"): self._get_feature_value(field_meta.get("dataType"), field_value)
+                for field_meta, field_value in zip(field_meta_list, raw_feature_value)
+            }
+        else:
+            value = raw_feature_value
+        return value
```
Comment (brian-tecton-ai) [inline tecton_client/_internal/data_types.py:76]: nit: a style I like better for this type of construct is returning early in the conditional (makes it clear that the result isn't b eing used later)

```
if value is None:
    return None
if data_type == "int64":
    return _convert_int64(raw_feature_value)
if data_type == "array":
    element_type = data_type_info.get("elementType")
    return [_get_feature_value(element_type, value) for value in raw_feature_value]
... etc
```

**Review by brian-tecton-ai — APPROVED** [review]

File: tecton_client/_internal/data_types.py (hunk 1 above)
Comment (solbrigm) [reply tecton_client/_internal/data_types.py:4]: Yeah I tested later they actually _aren't_ valid; it should only  only for Float64. Will remove.

File: tecton_client/_internal/data_types.py (hunk 2 above)
Comment (solbrigm) [reply tecton_client/_internal/data_types.py:50]: Will make the function static; that's equivalent, but still seems like it belongs in the GetFeaturesResponse namespace. Added tests.

**Pull request merged**